File: hadoop/src/main/scala/MaxClosePriceReducer.scala
package hadoop.studying

import org.apache.hadoop.mapreduce.Reducer
import org.apache.hadoop.io.{Text, FloatWritable}

public class MaxClosePriceReducer {

}
    extends Reducer[Text, FloatWritable, Text, FloatWritable] {
      override protected[mapreduce] def reduce(key: Text, values: lang.Iterable[FloatWritable], context: Context): Unit = ???
    }

File: hadoop/src/main/java/MaxClosePriceReducer.java
package hadoop.studying;

import org.apache.hadoop.mapreduce.Reducer;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.io.FloatWritable;
import java.io.IOException;
import java.lang.Iterable;

public class MaxClosePriceReducer
    extends Reducer<Text, FloatWritable, Text, FloatWritable> {
    @Override
    protected void reduce(Text key, Iterable<FloatWritable> values,  Context context)
      throws IOException, InterruptedException {
      Float max = Float.MIN_VALUE;
      for (FloatWritable value : values) {
        max = Math.max(max, value.get());
      }

      context.write(key, new FloatWritable(max));
    }
}
